tests/e2e/tidbcluster/serial.go

```diff
@@ -589,6 +589,245 @@ var _ = ginkgo.Describe("[Serial]", func() {
 		})
 	})
 
+	ginkgo.Describe("Canary Deploy Operator from 1.1.10 to latest", func() {
+		var oa *tests.OperatorActions
+		var ocfg *tests.OperatorConfig
+		var operatorVersion string
+
+		ginkgo.BeforeEach(func() {
+			operatorVersion = "v1.1.10"
+			ocfg = &tests.OperatorConfig{
+				Namespace:       ns,
+				ReleaseName:     "operator",
+				Tag:             operatorVersion,
+				Image:           fmt.Sprintf("pingcap/tidb-operator:%s", operatorVersion),
+				ImagePullPolicy: v1.PullIfNotPresent,
+			}
+			oa = tests.NewOperatorActions(cli, c, asCli, aggrCli, apiExtCli, tests.DefaultPollInterval, ocfg, e2econfig.TestConfig, nil, fw, f)
+			ginkgo.By("Installing CRDs")
+			oa.DeployReleasedCRDOrDie(operatorVersion)
+			ginkgo.By("Installing tidb-operator")
+			oa.CleanOperatorOrDie(ocfg)
+			oa.DeployOperatorOrDie(ocfg)
+		})
+
+		ginkgo.AfterEach(func() {
+			ginkgo.By("Uninstall tidb-operator")
+			oa.CleanOperatorOrDie(ocfg)
+			ginkgo.By("Uninstalling CRDs")
+			tests.CleanReleasedCRDOrDie(operatorVersion)
+		})
+
+		ginkgo.It("Deploy TidbCluster and check the result", func() {
+			ginkgo.By(fmt.Sprintf("deploy original tc %q", utilimage.TiDBV4Prev))
+			tcName := "tidbcluster"
+			tc := fixture.GetTidbCluster(ns, tcName, utilimage.TiDBV4Prev)
+			tc.Spec.PD.Replicas = 3
+			tc.Spec.TiKV.Replicas = 3
+			tc.Spec.TiDB.Replicas = 1
+
+			err := genericCli.Create(context.TODO(), tc)
+			framework.ExpectNoError(err, "Expected TiDB cluster created")
+			err = oa.WaitForTidbClusterReady(tc, 10*time.Minute, 5*time.Second)
+			framework.ExpectNoError(err, "Expected TiDB cluster ready")
+
+			getPods := func(ls string) ([]v1.Pod, error) {
+				listOptions := metav1.ListOptions{
+					LabelSelector: ls,
+				}
+				podList, err := c.CoreV1().Pods(ns).List(listOptions)
+				if err != nil {
+					return nil, err
+				}
+				return podList.Items, nil
+			}
+
+			tikvPods, err := getPods(labels.SelectorFromSet(label.New().Instance(tcName).TiKV().Labels()).String())
+			framework.ExpectNoError(err, "failed to get tidb pods")
```

Comment:
Suggest checking PD pods here as the PD will be upgraded before TiKV.

Reply:
fix